src/compiler/checker.ts

```diff
@@ -1672,6 +1672,14 @@ namespace ts {
                                 return [symbolFromSymbolTable];
                             }
 
+                            // If this is a "default" alias, use what it aliases instead.
+                            if (symbolFromSymbolTable.flags & SymbolFlags.Alias &&
+                                symbolFromSymbolTable.name === "default" &&
+                                symbolFromSymbolTable.declarations.length &&
+                                symbolFromSymbolTable.declarations[0].kind === SyntaxKind.ExportAssignment) {
+                                symbolFromSymbolTable = resolveAlias(symbolFromSymbolTable);

```

Comment:
the symbol `"default"` should never be used in qualifying anything. this is what this change does.